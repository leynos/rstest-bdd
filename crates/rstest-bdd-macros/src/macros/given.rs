//! Implementation of the `#[given]` macro.

use proc_macro::TokenStream;

/// Macro for defining a Given step that registers with the step inventory.
pub(crate) fn given(attr: TokenStream, item: TokenStream) -> TokenStream {
    super::step_attr(attr, item, crate::StepKeyword::Given)
}
