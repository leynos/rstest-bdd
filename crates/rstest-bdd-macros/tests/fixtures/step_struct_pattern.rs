use rstest_bdd_macros::given;

struct User {
    name: String,
}

#[given("user data")]
fn step_with_struct(User { name }: User) {}

fn main() {}
