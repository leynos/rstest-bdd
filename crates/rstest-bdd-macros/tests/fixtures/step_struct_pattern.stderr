error: invalid step function signature: unsupported parameter pattern; use a simple identifier (e.g., `arg: T`)
  --> tests/fixtures/step_struct_pattern.rs:11:21
   |
11 | fn step_with_struct(User { name }: User) {}
   |                     ^^^^^^^^^^^^^

error: help: use `#[given] fn name(ctx: &StepContext, ...)` and valid fixtures
  --> tests/fixtures/step_struct_pattern.rs:11:21
   |
11 | fn step_with_struct(User { name }: User) {}
   |                     ^^^^^^^^^^^^^
